### PR TITLE
`crmf_lib.c`: Make sure Edwards signature for POPO is called without digest

### LINKED ISSUE #18184: openssl 3.0.2 cmp -cmd ir doesn't work with ed25519 key
I would like to know if it's normal that with the key type ed25519, I got:

CMP:apps/cmp.c:1400:CMP warning: -cert and -key not used for protection since -secret is given
CMP:apps/cmp.c:1958:CMP info: will contact http://192.168.1.1:80/cmp_path
CMP error: invalid digest
CMP error: error creating certreq

It's working with the key type rsa . I don't get the message: CMP error: invalid digest

### PR DESCRIPTION
**Pull request by DDvO**
This replaces #18195.

Fixes #18184

### CODE REVIEW

File: crypto/asn1/a_sign.c [hunk 1]
```diff
@@ -131,12 +131,11 @@ int ASN1_item_sign_ex(const ASN1_ITEM *it, X509_ALGOR *algor1,
 {
     int rv = 0;
     EVP_MD_CTX *ctx = evp_md_ctx_new_ex(pkey, id, libctx, propq);
-
```
Comment (t8m) [inline crypto/asn1/a_sign.c:134]: Please just drop the changes in this file. It is unrelated now.

**Review by t8m — APPROVED** [review]

File: crypto/asn1/a_sign.c (hunk 1 above)
Comment (DDvO) [reply crypto/asn1/a_sign.c]: Done.

**Comment by t8m** [comment]
I could approve this one however did you consider doing it at `ossl_cmp_certreq_new()` instead? I still feel like this should be done as early as possible in the call stack.

**Comment by DDvO** [comment]
> I could approve this one however did you consider doing it at `ossl_cmp_certreq_new()` instead? I still feel like this should be done as early as possible in the call stack.

Good point to take this into consideration - I had not thought about this possibility so far.

I'd say `create_popo_signature()` is the right place because only here it is clear that the `digest` is going to be used for (self-)signature with the `pkey`.
When calling `OSSL_CRMF_MSG_create_popo()` from `cmp_msg.c`, this is not yet clear,
and (at least in the future) the `digest` might also be used for something else.

Nevertheless, these thoughts reminded me to update the doc of `OSSL_CRMF_MSG_create_popo()`.

**Comment by DDvO** [comment]
On this occasion also fixing a glitch in `OSSL_CRMF_MSG_set0_validity.pod`.

File: doc/man3/OSSL_CRMF_MSG_set0_validity.pod [hunk 2]
```diff
@@ -62,11 +62,13 @@ OSSL_CRMF_MSG_push0_extension() pushes the X509 extension I<ext> to the
...
 
 OSSL_CRMF_MSG_create_popo() creates and sets the Proof-of-Possession (POPO)
-according to the method I<meth> in I<crm>.<
+according to the method I<meth> in I<crm>.
 The library context I<libctx> and property query string I<propq>,
 may be NULL to select the defaults.
 In case the method is OSSL_CRMF_POPO_SIGNATURE the POPO is calculated
-using the private key I<pkey> and the digest method I<digest>.
+using the private key I<pkey> and the digest method I<digest>,
+where the I<digest> argument is ignored if I<pke> is of a type (such as
```
Comment (t8m) [inline doc/man3/OSSL_CRMF_MSG_set0_validity.pod:70]: Typo: pkey

Comment (DDvO) [reply doc/man3/OSSL_CRMF_MSG_set0_validity.pod]: Fixed

**Review by t8m — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by DDvO** [comment]
Squashed and merged to master and 3.0 -
thanks to all involved in discussing the topic.

**Pull request closed without merging**